test/conformance/user_test.go

```diff
@@ -0,0 +1,64 @@
+// +build e2e
+
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conformance
+
+import (
+	"testing"
+
+	"github.com/knative/serving/test"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const userID = 2020
+
+// TestMustRunAsUser verifies that a supplied runAsUser through securityContext takes
+// effect as delared by "MUST" in the runtime-contract.
+func TestMustRunAsUser(t *testing.T) {
+	t.Parallel()
+	clients := setup(t)
+
+	runAsUser := int64(userID)
+	securityContext := &corev1.SecurityContext{
+		RunAsUser: &runAsUser,
+	}
+
+	ri, err := fetchRuntimeInfo(t, clients, &test.Options{SecurityContext: securityContext})
```

Comment:
I wonder,
what if we actually create the service and pull all the options in a TestMain and then the tests just verify different pieces of the contract. Or, at least, they are all part of the same test suite with the same `TestMain`, that creates the service and tests just query the runtime info?

Reply:
I am happy to look into refactoring this, but would prefer to do it in a separate follow-up PR as it will involve changing the existing tests. I would like to keep this one to strictly be additive of the new test.